lib/salesforce/supporter_finder.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'salesforce/client_api'
+
+module Salesforce
+  class SupporterFinder
+    ID = :Id
+    SORT_FIELD = :First_entered__c
+
+    def self.execute(field, value, client = ClientAPI.new)
+      new(client, field, value).execute
+    end
+
+    def initialize(client, field, value)
+      @client = client
+      @field = field
+      @value = value
+    end
+
+    def execute
+      Result.new(supporter, errors)
+    end
+
+    private
+
+    attr_reader :client, :field, :value
+
+    def table
+      'Contact'
+    end
+
+    def supporter
+      first = first_entered
+      fetch(first[ID]) if first
+    end
+
+    def first_entered
+      search.sort_by { |item| item[SORT_FIELD] }.first if search
+    end
+
+    def expression
+      "select #{ID}, #{SORT_FIELD} from #{table} where #{field}='#{value}'"
```

Comment:
This also looks like a constant?

Reply:
we might have many of these SOQL expressions in the future... I don't know yet.